src/main/scala/com/databricks/labs/mosaic/core/geometry/MosaicGeometryESRI.scala

```diff
@@ -127,6 +127,8 @@ abstract class MosaicGeometryESRI(geom: OGCGeometry) extends MosaicGeometry {
 
     override def convexHull: MosaicGeometryESRI = MosaicGeometryESRI(geom.convexHull())
 
+    override def unaryUnion: MosaicGeometry = MosaicGeometryESRI(geom.union(geom))
```

Comment:
@tdikland I would add here a one liner to clarify why geom.union(geom) in ESRI is equivalent to geom.union() in JTS (I would argue JTS way is more intuitive).

Reply:
Completely agree. Added some inline explanations.